pandas/core/groupby/grouper.py

```diff
@@ -155,6 +162,8 @@ def _set_grouper(self, obj: FrameOrSeries, sort: bool = False):
             whether the resulting grouper should be sorted
         """
         assert obj is not None
+        if not obj.equals(self.obj):
```

Comment:
does `obj.equals(self.obj)` vs `obj is self.obj` make a difference?

Reply:
It shouldn't -- I'll update it